src/EditorFeatures/CSharpTest/QualifyMemberAccess/QualifyMemberAccessTests.cs

```diff
@@ -1143,5 +1143,23 @@ static Program()
 }",
 CodeStyleOptions.QualifyEventAccess);
         }
+
+        [WorkItem(21519, "https://github.com/dotnet/roslyn/issues/21519")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsQualifyMemberAccess)]
+        public async Task QualifyPropertyAccess_NameofArgument()
+        {
+            await TestAsyncWithOption(
+@"class Program
+{
+    public int Foo { get; set; }
+    public string Bar = nameof([|Foo|]);
+}",
+@"class Program
+{
+    public int Foo { get; set; }
+    public string Bar = nameof(this.Foo);
```

Comment:
@sharwell is on vacation and may not get a chance to look at this. @CyrusNajmabadi can you?